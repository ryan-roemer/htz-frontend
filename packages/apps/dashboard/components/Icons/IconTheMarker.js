import React from 'react';
import { FelaComponent, } from 'react-fela';
import iconStyle from './iconStyle';

const IconTheMarker = ({ size, fill, color, attrs, miscStyles, ...props }) => (
  <FelaComponent
    size={size}
    fill={fill}
    color={color}
    attrs={attrs}
    miscStyles={miscStyles}
    rule={iconStyle}
    render={({ className, }) => (
      <svg width="1em" height="1em" viewBox="0 0 256 256" className={className} {...props} {...attrs} >  <path fill="currentColor" d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z" /></svg>
    )}
  />
);

export default IconTheMarker;
