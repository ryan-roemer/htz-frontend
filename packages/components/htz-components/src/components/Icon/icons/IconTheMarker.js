/** ************************************************************** *
 * THIS IS AN AUTO GENERATED FILE. PLEASE DO NOT EDIT IT DIRECTLY.
 *
 * If you want to change this Icon component , it is generated
 * from the `iconTamplate.js` file the parent directory.
 * *************************************************************** */
import React from 'react';
import { createComponent, } from 'react-fela';
import iconStyle from '../iconStyle';
import { iconPropTypes, iconDefaultProps, } from '../iconPropTypes';
import { attrsPropType, } from '../../../propTypes/attrsPropType';

const IconTheMarker = createComponent(iconStyle, UnstyledIconTheMarker, [
  'attrs',
  'onClick',
]);

IconTheMarker.propTypes = iconPropTypes;
IconTheMarker.defaultProps = iconDefaultProps;

// Underlying component
UnstyledIconTheMarker.propTypes = {
  attrs: attrsPropType,
};

UnstyledIconTheMarker.defaultProps = {
  attrs: null,
};

function UnstyledIconTheMarker({ attrs, ...props }) {
  return (
    <svg width="1em" height="1em" viewBox="0 0 256 256" {...props} {...attrs}>
      {' '}
      <circle fill="currentColor" cx={128} cy={128} r={120} />{' '}
      <path d="M191 181h-29v-77l-20 77h-28l-19-77v77H66V75h48l14 60 14-60h49v106z" />
    </svg>
  );
}

export default IconTheMarker;
