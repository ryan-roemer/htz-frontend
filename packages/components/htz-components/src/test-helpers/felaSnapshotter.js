import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
