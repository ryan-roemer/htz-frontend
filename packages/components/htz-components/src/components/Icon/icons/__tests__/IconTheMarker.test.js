/* global window */
/** ************************************************************** *
 * THIS IS AN AUTO GENERATED FILE. PLEASE DO NOT EDIT IT DIRECTLY.
 *
 * If you want to change the styleguide example, it is generated
 * from the `embedsFileTemplate.js` file is this directory.
 * *************************************************************** */
import React from 'react';
import felaSnapshotter from '../../../../test-helpers/felaSnapshotter';
import IconTheMarker from '../IconTheMarker';

describe('<IconTheMarker />', () => {
  /* The error tests are left as documentation, but are commented out for the redability of the test results */
  // describe('Errors', () => {
  //   it('throw an error when the value passed to the "size" prop isn't a number', () => {
  //     expect(() => felaSnapshotter(<IconTheMarker size="2rem" />)).toThrow(
  //       /^An Icon's "size" prop may only be passed a "number"/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconTheMarker color="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconTheMarker color={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconTheMarker fill="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconTheMarker fill={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  // });
  describe('DOM element', () => {
    testCase('render correctly when no props are passed', <IconTheMarker />);
    testCase(
      'pass attributes defined in the "attrs" prop to the DOM element',
      <IconTheMarker attrs={{ 'aria-hidden': true, }} />
    );
    testCase('add "id" attribute to DOM element', <IconTheMarker id="hasId" />);
    testCase(
      'add "onClick" attribute to DOM element',
      <IconTheMarker id={e => console.log('a')} />
    );
  });

  describe('styles', () => {
    testCase("set icon's color", <IconTheMarker color={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's color responsively",
      <IconTheMarker
        color={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's fill", <IconTheMarker fill={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's fill responsively",
      <IconTheMarker
        fill={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's size", <IconTheMarker size={6} />);
    testCase(
      "set icon's size responsively",
      <IconTheMarker
        size={[ { until: 'xl', value: 6, }, { from: 'xl', value: 7, }, ]}
      />
    );
    testCase(
      'set misc styles on an icon',
      <IconTheMarker miscStyles={{ position: 'relative', }} />
    );
    testCase(
      'responsively set misc styles on an icon',
      <IconTheMarker
        miscStyles={{
          position: [
            { until: 'xl', value: 'relative', },
            { from: 'xl', value: 'static', },
          ],
        }}
      />
    );
  });
});

function testCase(name, Component, { only = false, } = {}) {
  const testFunc = only ? it.only : it;
  testFunc(name, () => {
    const { component, styles, } = felaSnapshotter(Component);
    expect(component).toMatchSnapshot();
    expect(styles).toMatchSnapshot();
  });
}
