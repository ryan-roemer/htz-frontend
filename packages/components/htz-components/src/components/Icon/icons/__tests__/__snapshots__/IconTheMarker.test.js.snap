// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconTheMarker /> DOM element add "id" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id="hasId"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> DOM element add "id" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconTheMarker /> DOM element add "onClick" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id={[Function]}
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> DOM element add "onClick" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconTheMarker /> DOM element pass attributes defined in the "attrs" prop to the DOM element 1`] = `
<div>
  <svg
    aria-hidden={true}
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> DOM element pass attributes defined in the "attrs" prop to the DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconTheMarker /> DOM element render correctly when no props are passed 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> DOM element render correctly when no props are passed 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconTheMarker /> styles responsively set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> styles responsively set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{position:relative}}@media (min-width: 80em){.htz-e{position:static}}"`;

exports[`<IconTheMarker /> styles set icon's color 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> styles set icon's color 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:#0B7EB5}.htz-c{fill:#FFF}"`;

exports[`<IconTheMarker /> styles set icon's color responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> styles set icon's color responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-d{fill:#FFF}@media (max-width: 79.9375em){.htz-b{color:#0B7EB5}}@media (min-width: 80em){.htz-c{color:#2D2D2D}}"`;

exports[`<IconTheMarker /> styles set icon's fill 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> styles set icon's fill 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#0B7EB5}"`;

exports[`<IconTheMarker /> styles set icon's fill responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> styles set icon's fill responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}@media (max-width: 79.9375em){.htz-c{fill:#0B7EB5}}@media (min-width: 80em){.htz-d{fill:#2D2D2D}}"`;

exports[`<IconTheMarker /> styles set icon's size 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> styles set icon's size 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{font-size:6rem}"`;

exports[`<IconTheMarker /> styles set icon's size responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> styles set icon's size responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{font-size:6rem}}@media (min-width: 80em){.htz-e{font-size:7rem}}"`;

exports[`<IconTheMarker /> styles set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 20C68.4 20 20 68.4 20 128s48.4 108 108 108 108-48.4 108-108S187.6 20 128 20zm56 155h-26v-68.9L140.5 175h-25.1L98 106.1V175H72V80h42.8l12.8 53.8 13-53.8H184v95z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconTheMarker /> styles set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{position:relative}"`;
